### ISSUE TITLE
H33 Supplementing link text with title attribute

### ISSUE THREAD

**Issue by vavroom** [w3c/wcag#305]
The information on https://www.w3.org/TR/WCAG20-TECHS/H33.html is good in theory, but in practice, the fact is that the title attribute is so poorly implemented that using title on links can cause more problems than it's worth.

At the very least there should be a mention of the potential issues related to poor implementation.

**Comment by karlgroves** [comment]
+1
I would go so far as to remove H33 altogether.

**Comment by alastc** [comment]
The technique already has a paragraph (the 2nd of 2) saying there are UA problems and pointing to better techniques to use.

I'll put up a question to the AG group about removing H33, but if there is a context (no matter how obscure) where it is still a relevent technique, it will probably not be removed.

**Comment by electronicwoft** [comment]
How does H33 sit with SC1.4.13 - content on focus or hover? I believe H64 should also be depracated and possibly also H65.

**Comment by alastc** [comment]
Hi @electronicwoft, SC1.4.13 has an exception for when: "The visual presentation of the additional content is controlled by the user agent and is not modified by the author." That covers title attributes.

**Comment by lauracarlson** [comment]
Related References:

## [Title attributes](http://simplyaccessible.com/article/title-attributes/) by Joanna Briggs
>  When hovering over a link that has a title attribute, the large mouse pointer covers the start of the title attribute. Longer title attributes may not fit inside the viewport with higher levels of magnification...In usability testing, we've even observed a user who was looking for a link on the screen. She didn’t know that she stopped her mouse over a link with a title attribute. That title attribute hid the link she was trying to hunt down. How do you avoid these problems? Just avoid using title attributes.

Simply Accessible has a [screenshot illustrating the problem](http://simplyaccessible.com/wordpress/wp-content/uploads/2013/09/400percentTitleAttribute.png). It shows a large hand cursor overlapping a tool tip that should read "2013 Fall TV Schedule". It reads "13 Fall TV Schedule". 

## [University of Minnesota Duluth Usability Study](https://www.w3.org/WAI/GL/low-vision-a11y-tf/wiki/index.php?title=Metadata_On_Hover&oldid=2426#University_of_Minnesota_Duluth) - UMD Accessibility Task Force

>UMD Students with low vision experienced serious issues with cursors overlapping tooltip text on hover during March 2015 formal usability testing of a teaching-and-learning Web product.

## [Title Texts Suck](https://axesslab.com/title-texts-suck/) by Hampus Sethfors
>"Many people I meet think title texts, also known as tooltips, improve both the accessibility and usability of their sites. They don’t. In fact, they can even cause problems...What should I use instead of title texts? Well, just make sure your link and button names clearly describe where they take your user and you’ll be fine. 

## [How-to: Use title attributes](https://a11yproject.com/posts/title-attributes/) by Dave Rupert
HTML title attributes are often perceived as an accessibility (and SEO) bonus, but the opposite is true. For screen reader users the content included inside of the title attribute is typically unnecessary, redundant, and possibly not even used. Conversely, content being put in the title attribute is being hidden from the (probable) majority of your users. If information is being hidden from the majority of your users, then it’s likely not necessary.

## [The Trials and Tribulations of the Title Attribute ](https://www.24a11y.com/2017/the-trials-and-tribulations-of-the-title-attribute/) by Scott O’Hara

> The title attribute has provided glimpses of its potential usefulness. But because of some bad practices and inconsistent support, many title attribute use cases are pointless at best—or actively harmful—to those who can access them. The continued lack of support from all browser vendors to reveal titles to non-mouse users, and just the fact that tooltips in general call out inadequacies with user interfaces, should deter their usage for more straightforward and always accessible content. - 

## [Using the HTML title attribute – updated ](https://developer.paciellogroup.com/blog/2013/01/using-the-html-title-attribute-updated/)  by Steve Faulkner 
>  "…With the rise of touch screen interfaces, the usefulness of this attribute has decreased. The accessibility of the title attribute has fallen victim to a unfortunate combination of poor browser support, poor screen reader support and poor authoring practices

## [HTML5 Specification](https://html.spec.whatwg.org/multipage/dom.html#the-title-attribute)
> Relying on the title attribute is currently discouraged as many user agents do not expose the attribute in an accessible manner as required by this specification (e.g., requiring a pointing device such as a mouse to cause a tooltip to appear, which excludes keyboard-only users and touch-only users, such as anyone with a modern phone or tablet). 

## [Karl Groves](https://twitter.com/karlgroves/status/941809410401275905)
> If I found a genie and it gave me one wish, it would be that web developers stopped using the title attribute

**Comment by alastc** [comment]
The AG group discussed this on Tuesday and came to the conclusion we should not remove H33. 

The [Meeting minutes](https://www.w3.org/2019/07/09-ag-minutes.html#item06) will give you a flavour, but the crux of it was that whilst the technique is not 'recommended', there are scenarios where it is valid to use. E.g. The design can't be touched, you need to apply some more description to certain links and there isn't another way of doing it.

If anyone wishes to put forward an update, e.g. providing a new technique that acheives the same thing, or link to other resources, or an update to this technique, that option is open. We'd just need someone to put a little time (and ideally a PR) in.
